src/components/ha-label-badge.ts

```diff
@@ -25,11 +27,23 @@ class HaLabelBadge extends LitElement {
     return html`
       <div class="badge-container">
         <div class="label-badge" id="badge">
+          ${this.upperLabel
+            ? html`
+                <div
+                  class=${classMap({
+                    upperLabel: true,
+                    big: this.upperLabel.length > 5,
+                  })}
+                >
+                  <span>${this.upperLabel}</span>
+                </div>
+              `
+            : ""}
           <div
-            class="${classMap({
+            class=${classMap({
               value: true,
```

Comment:
you didnt write this but value: true can just be the string in the class like above